combine.lua

```diff
@@ -0,0 +1,418 @@
+-- Combines food and plant items across stockpiles.
+local argparse = require('argparse')
+local utils = require('utils')
+
+local opts, args = {
+    help = false,
+    all = nil,
+    here = nil,
+    dryrun = false,
+    types = nil,
+    verbose = false
+  }, {...}
+
+  -- default max stack size of 30
+local DEF_MAX=30
+
+-- list of valid item types for merging
+local valid_types_map = {
+    ['all'] = { },
+    ['drink'] = {[df.item_type.DRINK]={type_id=df.item_type.DRINK, type_name='DRINK',type_caste=false,max_stack_size=DEF_MAX}},
+    ['fat'] =   {[df.item_type.GLOB]={type_id=df.item_type.GLOB, type_name='GLOB',type_caste=false,max_stack_size=DEF_MAX},
+                [df.item_type.CHEESE]={type_id=df.item_type.CHEESE, type_name='CHEESE',type_caste=false,max_stack_size=DEF_MAX}},
+    ['fish'] = {[df.item_type.FISH]={type_id=df.item_type.FISH, type_name='FISH',type_caste=true,max_stack_size=DEF_MAX},
+                [df.item_type.FISH_RAW]={type_id=df.item_type.FISH_RAW, type_name='FISH_RAW',type_caste=true,max_stack_size=DEF_MAX},
+                [df.item_type.EGG]={type_id=df.item_type.EGG, type_name='EGG',type_caste=true,max_stack_size=DEF_MAX}},
+    ['food'] = {[df.item_type.FOOD]={type_id=df.item_type.FOOD, type_name='FOOD',type_caste=false,max_stack_size=DEF_MAX}},
+    ['meat'] = {[df.item_type.MEAT]={type_id=df.item_type.MEAT, type_name='MEAT',type_caste=false,max_stack_size=DEF_MAX}},
+    ['plant'] = {[df.item_type.PLANT]={type_id=df.item_type.PLANT, type_name='PLANT',type_caste=false,max_stack_size=DEF_MAX},
+                [df.item_type.PLANT_GROWTH]={type_id=df.item_type.PLANT_GROWTH, type_name='PLANT_GROWTH',type_caste=false,max_stack_size=DEF_MAX}}
+}
+
+-- populate all types entry
+for k1,v1 in pairs(valid_types_map) do
+    if k1 ~= 'all' then
+        for k2,v2 in pairs(v1) do
+            valid_types_map['all'][k2]={}
+            for k3,v3 in pairs (v2) do
+                valid_types_map['all'][k2][k3]=v3
+            end
+        end
+    end
+end
+
+function log(...)
+    -- if verbose is specified, then print the arguments, or don't.
+    if opts.verbose then dfhack.print(string.format(...)) end
```

Comment:
not that you have to change anything, but if you use `print` here instead of `dfhack.print` then you won't have to end each of your log lines with `\n`